stats.py

```diff
@@ -0,0 +1,82 @@
+from weather import WeatherData
+
+class WeatherStats:
+    """
+    A class to calculate statistics from weather data.
+    Methods:
+        find_extremes(weather_data): Finds the highest temperature, lowest temperature, and most humid day.
+        calculate_averages(weather_data): Calculates the average highest temperature, lowest temperature, and mean humidity.
+    """
+    @staticmethod
+    def find_extremes(weather_data):
+        """
+        Finds the highest temperature, lowest temperature, and most humid day from the weather data.
+
+        Parameters:
+            weather_data (list): A list of WeatherData instances.
+
+        Returns:
+            tuple: A tuple containing the highest temperature, highest temperature day,
+                   lowest temperature, lowest temperature day, most humid day, and most humid day date.
+        """
+        extremes = {
+            "highest_temp": WeatherData(date="", max_temp=None, min_temp=None, mean_humidity=None),
+            "lowest_temp": WeatherData(date="", max_temp=None, min_temp=None, mean_humidity=None),
+            "most_humid": WeatherData(date="", max_temp=None, min_temp=None, mean_humidity=None)
+        }
+
+        for data in weather_data:
+            if data.max_temp is not None:
+                if extremes["highest_temp"].max_temp is None or data.max_temp > extremes["highest_temp"].max_temp:
+                    extremes["highest_temp"] = data
```

Comment:
Can be rewritten as if data.max_temp and (extremes["highest_temp"].max_temp is None or data.max_temp > extremes["highest_temp"].max_temp):

Reply:
Changing it to this instead:
            if data.max_temp is not None and (
                extremes["highest_temp"].max_temp is None
                or data.max_temp > extremes["highest_temp"].max_temp
            ):
as skipping (is not None) will evaluate a minimum temperature of 0 in the files as falsy and not evaluate it:
            if data.min_temp and (
                extremes["lowest_temp"].min_temp is None
                or data.min_temp < extremes["lowest_temp"].min_temp
            ):